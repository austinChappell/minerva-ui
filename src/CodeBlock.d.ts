/// <reference types="react" />
export declare const liveEditorStyle: {
  fontSize: number;
  marginBottom: number;
  marginTop: number;
  fontFamily: string;
  borderRadius: number;
  padding: number;
};
export declare const liveErrorStyle: {
  fontFamily: string;
  fontSize: number;
  padding: string;
  color: string;
  backgroundColor: string;
};
declare const CodeBlock: {
  ({ className, live, isManual, render, children, ...props }: any): JSX.Element;
  defaultProps: {
    mountStylesheet: boolean;
  };
  displayName: string;
  __docgenInfo: {
    description: string;
    displayName: string;
    props: {};
  };
};

declare module '@mdx-js/react';

export default CodeBlock;
