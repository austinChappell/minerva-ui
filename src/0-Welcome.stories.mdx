<Meta title="About|Introduction" />

# Get started

Minerva UI is a reusable component library to help build UIs faster. This library aims to be highly composable, declarative and accessible.

# Install
`npm install --save minerva-ui`
or
`yarn add minerva-ui`

# Usage

Import components you want into your UI:

```js
import { Checkbox } from 'minerva-ui';
```

And use them:

```jsx
() => {
  const [checked, setChecked] = React.useState(false);
  return (
    <Checkbox
      checked={checked}
      onChange={() => setChecked(!checked)}
    >
      Stay Logged In
    </Checkbox>
  )
}
```

# Utility Props

Utility props are provided as aliases for most components. The style is heavily influenced by [Tailwind CSS](https://tailwindcss.com/docs/font-size).

For example:
```jsx
() => (
  <>
    {/* enter a custom pixel value */}
    <Button fontSize="12px">Save</Button>
    {/* or enter the value named "sm" in the theme, which is .875rem by default */}
    <Button fontSize="lg">Save</Button>
  </>
)
```

# Styling Props Example

Here's the ["Card" example](https://tailwindcss.com/components/cards) by Tailwind recreated using this API:

```jsx
() => (
  <Flex
    flexDirection="column"
    boxShadow="lg"
    borderRadius="md"
    maxWidth="24rem"
    overflow="hidden"
  >
    <img
      src="https://tailwindcss.com/img/card-top.jpg"
      alt="Sunset in the mountains"
    />
    <Flex flexDirection="column" px={6} py={4}>
      <Text fontWeight="bold" fontSize="xl" mb={2} color="gray.700">The Coldest Sunset</Text>
      <Text color="gray.700" lineHeight="normal">Lorem ipsum dolor sit amet, consectetur adipisicing elit. Voluptatibus quia, nulla! Maiores et perferendis eaque, exercitationem praesentium nihil.</Text>
    </Flex>
    <Flex px={6} py={4}>
      <Block color="gray.700" bg="gray.200" borderRadius="full" px={3} py={1} mr={2}>
        #photography
      </Block>
      <Block color="gray.700" bg="gray.200" borderRadius="full" px={3} py={1} mr={2}>
        #travel
      </Block>
      <Block color="gray.700" bg="gray.200" borderRadius="full" px={3} py={1}>
        #winter
      </Block>
    </Flex>
  </Flex>
)
```
<!-- ### Styling
To include base styling, you can import the styles for the corresponding component:
```
import 'minerva-ui/checkbox.css'
``` -->