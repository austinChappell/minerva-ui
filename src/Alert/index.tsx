import React, { forwardRef } from 'react';
import styled from 'styled-components';
import { MinervaProps, systemProps, Flex } from '../layout';
import Icon from '../Icon';

const StyledAlert = styled(Flex)(
  props => ({
    padding: '16px',
    borderRadius: '5px',
  }),
  systemProps
);

const StyledAlertTitle = styled('p')(
  {
    fontWeight: 'bold',
    margin: '0 5px 0 8px',
  },
  systemProps
);

const StyledAlertDescription = styled.p``;

export interface AlertProps extends MinervaProps {
  children?: React.ReactNode;
  status?: 'error' | 'success' | 'warning' | 'info';
  title?: string;
  description?: string;
  color?: string;
  icon?: string;
  props?: any;
}

// @TODO: Move these to the theme so they can be customized
const alertTypes = {
  error: {
    statusColor: '#f8b4b4',
    statusIcon: 'alert-circle',
  },
  success: {
    statusColor: '#bcf0da',
    statusIcon: 'check-circle',
  },
  warning: {
    statusColor: '#fdf6b2',
    statusIcon: 'alert-triangle',
  },
  info: {
    statusColor: '#c3ddfd',
    statusIcon: 'info',
  },
};

export const Alert = forwardRef(function Alert(
  { title, description, status, color, icon, ...props }: AlertProps,
  ref
) {
  const { statusColor, statusIcon } =
    // if status is provided and is a provided alertType, return it
    status && Object.keys(alertTypes).includes(status)
      ? alertTypes[status]
      : {
          statusColor: '#e5e7eb',
          statusIcon: null,
        };

  const alertIcon = icon || statusIcon;

  return (
    <StyledAlert
      ref={ref}
      backgroundColor={color ? color : statusColor}
      {...props}
    >
      {alertIcon && <Icon name={alertIcon} size="20px" />}
      <StyledAlertTitle>{title}</StyledAlertTitle>
      <StyledAlertDescription>{description}</StyledAlertDescription>
    </StyledAlert>
  );
});

export default Alert;
