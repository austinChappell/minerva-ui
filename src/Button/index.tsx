import React, { forwardRef } from 'react';
import Spinner from '../Spinner';
import PseudoBox, { PseudoBoxProps } from '../PseudoBox';
import { MinervaProps, systemProps } from '../layout';
// import { useTheme } from '../theme';
import styled from 'styled-components';

const StyledButton = styled(PseudoBox)(
  props => ({
    ...props.theme.Button,
  }),
  systemProps
);

export interface ButtonProps extends MinervaProps, PseudoBoxProps {
  children?: React.ReactNode;
  /** Toggles disabled pseudo class */
  disabled?: boolean;
  /** If `true`, button will show a spinner. */
  isLoading?: boolean;
}

export const Button = forwardRef(function Button(
  {
    children,
    disabled = false,
    as: Comp = 'button',
    isLoading = false,
    ...props
  }: ButtonProps,
  ref
) {
  // const theme = useTheme();

  return (
    <StyledButton
      ref={ref}
      as={Comp}
      disabled={disabled || isLoading}
      role="button"
      transition="all 150ms ease 0s"
      outline="none"
      _hover={{
        backgroundColor: '#f9fafb',
      }}
      _focus={{
        borderColor: '#a4cafe',
        boxShadow: '0 0 0 3px rgba(118,169,250,.45)',
        outline: 0,
      }}
      _disabled={{
        opacity: 0.4,
        cursor: 'not-allowed',
      }}
      // {...theme.Button}
      {...props}
    >
      {isLoading ? <Spinner /> : children}
    </StyledButton>
  );
});

export default Button;
