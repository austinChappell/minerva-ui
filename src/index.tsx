/**
 * Utilities
 */

export { default as ThemeProvider } from './ThemeProvider';
export { default as GlobalStyles } from './GlobalStyles';
export { default as defaultTheme } from './theme';
export { default as styled } from 'styled-components';

/**
 * Display
 */
export { default as Text } from './Text';
export { default as Link } from './Link';
export { default as Heading } from './Heading';
export { default as Tag } from './Tag';
export { default as Alert } from './Alert';

/**
 * Layout
 */
export { Box, Block, Flex } from './layout';
export { default as PseudoBox } from './PseudoBox';
export { default as Stack } from './Stack';

/**
 * Inputs
 */
export { default as Button } from './Button';
export { default as Checkbox } from './Checkbox';
export { default as Input } from './Input';
export { RadioGroup, Radio } from './Radio';
export { default as InputField } from './InputField';
export { default as Select } from './Select';
export { default as TagsInput } from './TagsInput';

/**
 * Elements
 */
export { default as Image } from './Image';
export { default as Table } from './Table';
export { default as Icon } from './Icon';
export * from './Table';
export { default as Skeleton } from './Skeleton';

/**
 * Helpers
 */
export { default as Spinner } from './Spinner';
