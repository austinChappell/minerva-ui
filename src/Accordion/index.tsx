import React, { forwardRef } from 'react';
import styled from 'styled-components';
import {
  Accordion as ReachAccordion,
  AccordionItem as ReachAccordionItem,
  AccordionButton as ReachAccordionButton,
  AccordionPanel as ReachAccordionPanel,
  AccordionProps as ReachAccordionProps,
} from '@reach/accordion';
import { MinervaProps, Box, systemProps } from '../layout';
import Icon from '../Icon';
import Button from '../Button';

export interface AccordionProps extends MinervaProps {
  children?: React.ReactNode;
  props?: any;
}

export interface AccordionItemProps extends MinervaProps {
  children?: React.ReactNode;
  props?: any;
}

export interface AccordionButtonProps extends MinervaProps {
  children?: React.ReactNode;
  props?: any;
}

export interface AccordionPanelProps extends MinervaProps {
  children?: React.ReactNode;
  props?: any;
}

type AllAccordionProps = ReachAccordionProps & MinervaProps;
export const Accordion = styled(ReachAccordion)<AllAccordionProps>(
  {
    borderBottomColor: 'gray.200',
    borderBottomWidth: '1px',
  },
  systemProps
);

export const AccordionItem = forwardRef(function AccordionItem(
  { ...props }: AccordionItemProps,
  ref
) {
  return <Box as={ReachAccordionItem} ref={ref} {...props} />;
});

export const AccordionButton = forwardRef(function AccordionButton(
  { children, ...props }: AccordionButtonProps,
  ref
) {
  return (
    <Button
      as={ReachAccordionButton}
      borderColor="transparent"
      width="100%"
      flexDirection="row"
      justifyContent="space-between"
      alignItems="flex-start"
      bg="white"
      py="12px"
      fontSize="18px"
      fontWeight="400"
      borderTopColor="gray.200"
      borderTopWidth="1px"
      ref={ref}
      {...props}
    >
      {children}
      {/* TODO Need to figure out how to determine if Item is expanded and rotate icon */}
      {/* <Icon name="chevron-down" ml={2} size="14px" /> */}
    </Button>
  );
});

export const AccordionPanel = forwardRef(function AccordionPanel(
  { ...props }: AccordionPanelProps,
  ref
) {
  return (
    <Box
      as={ReachAccordionPanel}
      px="14px"
      py="14px"
      width="95%"
      ref={ref}
      {...props}
    />
  );
});

export const MainAccordion = forwardRef(function Alert(
  { children, ...props }: AccordionProps,
  ref
) {
  return (
    <Accordion {...props} ref={ref}>
      {children}
    </Accordion>
  );
});

export default MainAccordion;
