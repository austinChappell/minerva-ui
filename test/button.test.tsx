import React from 'react';
import { matchers } from 'jest-emotion';
import { render } from '@testing-library/react';
import { Button } from '../src';
import { ThemeProvider } from '../src';

expect.extend(matchers);

describe('<Button />', () => {
  it('should render', () => {
    const { container } = render(
      <ThemeProvider>
        <Button />
      </ThemeProvider>
    );
    expect(container).toMatchSnapshot();
  });

  it('should display text', () => {
    const content = 'Hello';
    const { container } = render(
      <ThemeProvider>
        <Button>{content}</Button>
      </ThemeProvider>
    );

    expect(container).toHaveTextContent(content);
  });

  it('should show default background color', () => {
    const { getByRole } = render(
      <ThemeProvider>
        <Button />
      </ThemeProvider>
    );

    const button = getByRole('button');

    expect(button).toHaveStyleRule('background-color', 'rgb(56,161,105)');
  });

  it('should show new background color if provided in theme', () => {
    const newColor = 'green';
    const theme = {
      Button: {
        container: {
          backgroundColor: newColor,
        },
      },
    };
    const { getByRole } = render(
      <ThemeProvider theme={theme}>
        <Button />
      </ThemeProvider>
    );

    const button = getByRole('button');

    expect(button).toHaveStyleRule('background-color', newColor);
  });
});
