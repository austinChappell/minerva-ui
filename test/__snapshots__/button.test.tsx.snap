// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should change style for media queries 1`] = `
.c0 {
  -webkit-transition: all 150ms ease 0s;
  transition: all 150ms ease 0s;
  outline: none;
  background-color: #fff;
  -webkit-transition: all 150ms ease 0s;
  transition: all 150ms ease 0s;
  outline: none;
  background-color: #fff;
  background-color: #fff;
  border-width: 1px;
  color: #374151;
  font-weight: 500;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-appearance: none;
  -webkit-box-align: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  position: relative;
  white-space: nowrap;
  vertical-align: middle;
  font-size: 14px;
  line-height: 20px;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  border-radius: 5px;
  -webkit-transition: all 150ms ease 0s;
  transition: all 150ms ease 0s;
  outline: none;
  background-color: #fff;
}

.c0:hover {
  background-color: #f9fafb;
}

.c0:focus {
  border-color: #a4cafe;
  box-shadow: 0 0 0 3px rgba(118,169,250,.45);
  outline: 0;
}

.c0:disabled,
.c0:disabled:focus,
.c0:disabled:hover,
.c0[aria-disabled=true],
.c0[aria-disabled=true]:focus,
.c0[aria-disabled=true]:hover {
  opacity: 0.4;
  cursor: not-allowed;
}

@media screen and (min-width:40em) {
  .c0 {
    background-color: #000;
  }
}

@media screen and (min-width:40em) {
  .c0 {
    background-color: #000;
  }

  .c0 {
    background-color: #000;
  }

  .c0 {
    background-color: #000;
  }

  .c0 {
    background-color: #000;
  }
}

@media screen and (min-width:40em) {
  .c0 {
    background-color: #000;
  }
}

<div>
  <button
    class="sc-AxirZ sc-AxhUy c0"
    role="button"
  >
    Responsive Button
  </button>
</div>
`;

exports[`<Button /> should change style if disabled 1`] = `
.c0 {
  -webkit-transition: all 150ms ease 0s;
  transition: all 150ms ease 0s;
  outline: none;
  -webkit-transition: all 150ms ease 0s;
  transition: all 150ms ease 0s;
  outline: none;
  background-color: #fff;
  border-width: 1px;
  color: #374151;
  font-weight: 500;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-appearance: none;
  -webkit-box-align: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  position: relative;
  white-space: nowrap;
  vertical-align: middle;
  font-size: 14px;
  line-height: 20px;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  border-radius: 5px;
  -webkit-transition: all 150ms ease 0s;
  transition: all 150ms ease 0s;
  outline: none;
}

.c0 {
  -webkit-transition: all 150ms ease 0s;
  transition: all 150ms ease 0s;
  outline: none;
  -webkit-transition: all 150ms ease 0s;
  transition: all 150ms ease 0s;
  outline: none;
  background-color: #fff;
  border-width: 1px;
  color: #374151;
  font-weight: 500;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-appearance: none;
  -webkit-box-align: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  position: relative;
  white-space: nowrap;
  vertical-align: middle;
  font-size: 14px;
  line-height: 20px;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  border-radius: 5px;
  -webkit-transition: all 150ms ease 0s;
  transition: all 150ms ease 0s;
  outline: none;
}

.c0 {
  -webkit-transition: all 150ms ease 0s;
  transition: all 150ms ease 0s;
  outline: none;
  -webkit-transition: all 150ms ease 0s;
  transition: all 150ms ease 0s;
  outline: none;
  background-color: #fff;
  border-width: 1px;
  color: #374151;
  font-weight: 500;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-appearance: none;
  -webkit-box-align: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  position: relative;
  white-space: nowrap;
  vertical-align: middle;
  font-size: 14px;
  line-height: 20px;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  border-radius: 5px;
  -webkit-transition: all 150ms ease 0s;
  transition: all 150ms ease 0s;
  outline: none;
}

.c0 {
  -webkit-transition: all 150ms ease 0s;
  transition: all 150ms ease 0s;
  outline: none;
  -webkit-transition: all 150ms ease 0s;
  transition: all 150ms ease 0s;
  outline: none;
  background-color: #fff;
  border-width: 1px;
  color: #374151;
  font-weight: 500;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-appearance: none;
  -webkit-box-align: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  position: relative;
  white-space: nowrap;
  vertical-align: middle;
  font-size: 14px;
  line-height: 20px;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  border-radius: 5px;
  -webkit-transition: all 150ms ease 0s;
  transition: all 150ms ease 0s;
  outline: none;
}

.c0:hover {
  background-color: #f9fafb;
}

.c0:focus {
  border-color: #a4cafe;
  box-shadow: 0 0 0 3px rgba(118,169,250,.45);
  outline: 0;
}

.c0:disabled,
.c0:disabled:focus,
.c0:disabled:hover,
.c0[aria-disabled=true],
.c0[aria-disabled=true]:focus,
.c0[aria-disabled=true]:hover {
  opacity: 0.4;
  cursor: not-allowed;
}

<div>
  <button
    class="sc-AxirZ sc-AxhUy c0"
    disabled=""
    role="button"
  >
    Disabled Button
  </button>
</div>
`;

exports[`<Button /> should contain spinner with \`isLoading\` prop 1`] = `
.c1 {
  color: #777;
  display: block;
  display: inline-block;
  border-bottom-color: transparent;
  border-left-color: transparent;
  border-right-color: currentColor;
  border-top-color: currentColor;
  border-radius: 9999px;
  color: #777;
  border-style: solid;
  height: 20px;
  width: 20px;
  border-width: 2px;
  margin: 2px;
  -webkit-animation: cilQsd 0.45s linear infinite;
  animation: cilQsd 0.45s linear infinite;
}

.c0 {
  -webkit-transition: all 150ms ease 0s;
  transition: all 150ms ease 0s;
  outline: none;
  -webkit-transition: all 150ms ease 0s;
  transition: all 150ms ease 0s;
  outline: none;
  background-color: #fff;
  border-width: 1px;
  color: #374151;
  font-weight: 500;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-appearance: none;
  -webkit-box-align: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  position: relative;
  white-space: nowrap;
  vertical-align: middle;
  font-size: 14px;
  line-height: 20px;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  border-radius: 5px;
  -webkit-transition: all 150ms ease 0s;
  transition: all 150ms ease 0s;
  outline: none;
}

.c0 {
  -webkit-transition: all 150ms ease 0s;
  transition: all 150ms ease 0s;
  outline: none;
  -webkit-transition: all 150ms ease 0s;
  transition: all 150ms ease 0s;
  outline: none;
  background-color: #fff;
  border-width: 1px;
  color: #374151;
  font-weight: 500;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-appearance: none;
  -webkit-box-align: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  position: relative;
  white-space: nowrap;
  vertical-align: middle;
  font-size: 14px;
  line-height: 20px;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  border-radius: 5px;
  -webkit-transition: all 150ms ease 0s;
  transition: all 150ms ease 0s;
  outline: none;
}

.c0 {
  -webkit-transition: all 150ms ease 0s;
  transition: all 150ms ease 0s;
  outline: none;
  -webkit-transition: all 150ms ease 0s;
  transition: all 150ms ease 0s;
  outline: none;
  background-color: #fff;
  border-width: 1px;
  color: #374151;
  font-weight: 500;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-appearance: none;
  -webkit-box-align: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  position: relative;
  white-space: nowrap;
  vertical-align: middle;
  font-size: 14px;
  line-height: 20px;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  border-radius: 5px;
  -webkit-transition: all 150ms ease 0s;
  transition: all 150ms ease 0s;
  outline: none;
}

.c0 {
  -webkit-transition: all 150ms ease 0s;
  transition: all 150ms ease 0s;
  outline: none;
  -webkit-transition: all 150ms ease 0s;
  transition: all 150ms ease 0s;
  outline: none;
  background-color: #fff;
  border-width: 1px;
  color: #374151;
  font-weight: 500;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-appearance: none;
  -webkit-box-align: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  position: relative;
  white-space: nowrap;
  vertical-align: middle;
  font-size: 14px;
  line-height: 20px;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  border-radius: 5px;
  -webkit-transition: all 150ms ease 0s;
  transition: all 150ms ease 0s;
  outline: none;
}

.c0:hover {
  background-color: #f9fafb;
}

.c0:focus {
  border-color: #a4cafe;
  box-shadow: 0 0 0 3px rgba(118,169,250,.45);
  outline: 0;
}

.c0:disabled,
.c0:disabled:focus,
.c0:disabled:hover,
.c0[aria-disabled=true],
.c0[aria-disabled=true]:focus,
.c0[aria-disabled=true]:hover {
  opacity: 0.4;
  cursor: not-allowed;
}

<div>
  <button
    class="sc-AxirZ sc-AxhUy c0"
    disabled=""
    role="button"
  >
    <div
      class="sc-AxirZ sc-AxiKw c1"
      color="#777"
    />
  </button>
</div>
`;

exports[`<Button /> should render 1`] = `
.c0 {
  -webkit-transition: all 150ms ease 0s;
  transition: all 150ms ease 0s;
  outline: none;
  -webkit-transition: all 150ms ease 0s;
  transition: all 150ms ease 0s;
  outline: none;
  background-color: #fff;
  border-width: 1px;
  color: #374151;
  font-weight: 500;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-appearance: none;
  -webkit-box-align: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  position: relative;
  white-space: nowrap;
  vertical-align: middle;
  font-size: 14px;
  line-height: 20px;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  border-radius: 5px;
  -webkit-transition: all 150ms ease 0s;
  transition: all 150ms ease 0s;
  outline: none;
}

.c0 {
  -webkit-transition: all 150ms ease 0s;
  transition: all 150ms ease 0s;
  outline: none;
  -webkit-transition: all 150ms ease 0s;
  transition: all 150ms ease 0s;
  outline: none;
  background-color: #fff;
  border-width: 1px;
  color: #374151;
  font-weight: 500;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-appearance: none;
  -webkit-box-align: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  position: relative;
  white-space: nowrap;
  vertical-align: middle;
  font-size: 14px;
  line-height: 20px;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  border-radius: 5px;
  -webkit-transition: all 150ms ease 0s;
  transition: all 150ms ease 0s;
  outline: none;
}

.c0 {
  -webkit-transition: all 150ms ease 0s;
  transition: all 150ms ease 0s;
  outline: none;
  -webkit-transition: all 150ms ease 0s;
  transition: all 150ms ease 0s;
  outline: none;
  background-color: #fff;
  border-width: 1px;
  color: #374151;
  font-weight: 500;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-appearance: none;
  -webkit-box-align: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  position: relative;
  white-space: nowrap;
  vertical-align: middle;
  font-size: 14px;
  line-height: 20px;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  border-radius: 5px;
  -webkit-transition: all 150ms ease 0s;
  transition: all 150ms ease 0s;
  outline: none;
}

.c0 {
  -webkit-transition: all 150ms ease 0s;
  transition: all 150ms ease 0s;
  outline: none;
  -webkit-transition: all 150ms ease 0s;
  transition: all 150ms ease 0s;
  outline: none;
  background-color: #fff;
  border-width: 1px;
  color: #374151;
  font-weight: 500;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-appearance: none;
  -webkit-box-align: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  position: relative;
  white-space: nowrap;
  vertical-align: middle;
  font-size: 14px;
  line-height: 20px;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  border-radius: 5px;
  -webkit-transition: all 150ms ease 0s;
  transition: all 150ms ease 0s;
  outline: none;
}

.c0:hover {
  background-color: #f9fafb;
}

.c0:focus {
  border-color: #a4cafe;
  box-shadow: 0 0 0 3px rgba(118,169,250,.45);
  outline: 0;
}

.c0:disabled,
.c0:disabled:focus,
.c0:disabled:hover,
.c0[aria-disabled=true],
.c0[aria-disabled=true]:focus,
.c0[aria-disabled=true]:hover {
  opacity: 0.4;
  cursor: not-allowed;
}

<div>
  <button
    class="sc-AxirZ sc-AxhUy c0"
    role="button"
  >
    Test
  </button>
</div>
`;
